server/services/omnichannel-voip/service.ts

```diff
@@ -17,12 +27,85 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 
 	private users: UsersRaw;
 
+	private voipRoom: VoipRoomsRaw;
+
 	constructor(db: Db) {
 		super();
 		this.users = new UsersRaw(db.collection('users'));
+		this.voipRoom = new VoipRoomsRaw(db.collection('rocketchat_room'));
 		this.logger = new Logger('OmnichannelVoipService');
 	}
 
+	private normalizeAgent(agentId: string): any {
+		if (!agentId) {
+			return;
+		}
+
+		if (!settings.get('Livechat_show_agent_info')) {
+			return { hiddenInfo: true };
+		}
+
+		const agent = Users.getAgentInfo(agentId);
+		const { customFields: agentCustomFields, ...extraData } = agent;
+
+		return Object.assign(extraData);
+	}
+
+	private async createVoipRoom(rid: string, name: string, agent: any, guest: ILivechatVisitor): Promise<string> {
+		check(rid, String);
+		check(name, String);
+		check(
+			guest,
+			Match.ObjectIncluding({
+				_id: String,
+				username: String,
+				status: Match.Maybe(String),
+				department: Match.Maybe(String),
+			}),
+		);
+
+		const { _id, username, department: departmentId, status = 'online' } = guest;
+		const newRoomAt = new Date();
+
+		this.logger.debug(`Creating livechat room for visitor ${_id}`);
+
+		const room = Object.assign({
+			_id: rid,
+			msgs: 0,
+			usersCount: 1,
+			lm: newRoomAt,
+			fname: name,
+			t: 'v',
+			ts: newRoomAt,
+			departmentId,
+			v: {
+				_id,
+				username,
+				token: guest.token,
+				status,
+			},
+			servedBy: {
+				_id: agent.agentId,
+				ts: new Date(),
+				username: agent.username,
+			},
+			cl: false,
+			open: true,
+			waitingResponse: true,
+			// this should be overriden by extraRoomInfo when provided
+			// in case it's not provided, we'll use this "default" type
+			source: {
+				type: 'voip',
+				alias: 'unknown',
+			},
+			queuedAt: newRoomAt,
+		});
+		const result = await this.voipRoom.insertOne(room);
+
+		const roomId = await result.insertedId;
+		return roomId;
```

Comment:
Instead of assigning roomId, return directly `result.insertedId`
```suggestion
		return result.insertedId;
```

Reply:
Fixed.